### PR TITLE
Revert "Remove KeyPolicy default in aws-kms-key.json (#63)" 

### PR DESCRIPTION
**Pull request by jbvengala**
*Issue #, if available:*

*Description of changes:*

This pull request reverts commit 6cd611742decb007ffa766b9b49e9c0a0f239871 as it is causing contract tests failures with https://github.com/aws-cloudformation/resource-schema-guard-rail/blob/main/docs/BREAKING_CHANGE.md PR001

Updated `tagProperty` in aws-kms-key.json to fix https://github.com/aws-cloudformation/resource-schema-guard-rail/blob/main/docs/BASIC_LINTING.md TAG010

This also adds logic to ignore the default key policy from aws-kms-key.json as it is causing Integration test failure.

### CODE REVIEW

File: common/src/main/java/software/amazon/kms/common/KeyTranslator.java [hunk 1]
```diff
@@ -39,6 +40,24 @@ public abstract class KeyTranslator<M> {
     protected static final String DEFAULT_POLICY_NAME = "default";
     protected static final int LIST_KEYS_PAGE_SIZE = 50;
 
+    /** This is the default key policy from aws-kme-key.json */
```
Comment (sruthisrip) [inline common/src/main/java/software/amazon/kms/common/KeyTranslator.java:43]: aws-kms-key.json

Comment (jtcul) [reply common/src/main/java/software/amazon/kms/common/KeyTranslator.java:43]: +1

**Review by jtcul — APPROVED** [review]

File: key/src/main/java/software/amazon/kms/key/ModelAdapter.java [hunk 2]
```diff
@@ -81,14 +84,17 @@ public static ResourceModel setDefaults(final ResourceModel model,
     public static String translatePolicyInput(final Object policy) {
```
[the remark this answers is not in the document]
Comment (jbvengala) [reply key/src/main/java/software/amazon/kms/key/ModelAdapter.java:84]: I added a unit test in UpdateHandlerTest which corresponds to this

File: key/src/main/java/software/amazon/kms/key/ModelAdapter.java [hunk 3]
```diff
@@ -81,14 +84,17 @@ public static ResourceModel setDefaults(final ResourceModel model,
...
-                return new ObjectMapper().writeValueAsString(policy);
+                policyString = new ObjectMapper().writeValueAsString(policy);
             } catch (final JsonProcessingException e) {
                 throw new TerminalException(e);
             }
+        } else {
+            policyString = (String) policy;
         }
-        return (String) policy;
+        return (Objects.equals(policyString, DEFAULT_KEY_POLICY_FROM_JSON)) ? "" : policyString;
```
[the remark this answers is not in the document]
Comment (jbvengala) [reply key/src/main/java/software/amazon/kms/key/ModelAdapter.java:97]: Ack

File: common/src/test/java/software/amazon/kms/common/KeyTranslatorTest.java [hunk 4]
```diff
@@ -49,8 +49,11 @@ public void testTranslatePolicyInputJsonProcessingException() throws JsonProcess
             .thenThrow(JsonProcessingException.class);
         final MockKeyTranslator mockKeyTranslator = new MockKeyTranslator(mockMapper);
 
-        assertThatExceptionOfType(TerminalException.class).isThrownBy(
-            () -> mockKeyTranslator.translatePolicyInput(TestConstants.DESERIALIZED_KEY_POLICY));
+        try {
+            mockKeyTranslator.translatePolicyInput(TestConstants.DESERIALIZED_KEY_POLICY);
```
[the remark this answers is not in the document]
Comment (jbvengala) [reply common/src/test/java/software/amazon/kms/common/KeyTranslatorTest.java:53]: I will add tests for that

**Review by pdz8 — APPROVED** [review]

**Pull request merged**